native/acter/src/api/device.rs

```diff
@@ -98,18 +98,16 @@ impl DeviceController {
 
         DeviceController {
             event_tx,
-            event_rx: Arc::new(Mutex::new(Some(event_rx))),
+            event_rx: Arc::new(event_rx),
             listener: Arc::new(listener),
         }
     }
 }
 
 impl Client {
-    pub fn device_event_rx(&self) -> Option<Receiver<DeviceEvent>> {
-        match self.device_controller.event_rx.try_lock() {
-            Ok(mut r) => r.take(),
-            Err(e) => None,
-        }
+    pub fn device_event_rx(&self) -> impl Stream<Item = DeviceEvent> {
+        BroadcastStream::new(self.device_controller.event_rx.resubscribe())
+            .map(|o| o.unwrap_or_default())
```

Comment:
this should _not_ create a default even when there is none, this is wrong. Please rewrite this to also use `filter_map`

Reply:
Fixed by https://github.com/acterglobal/a3/pull/2411/commits/65f1e301d3f52063f81cebb6fa60450eade7ebcc